### PR TITLE
[DPC-4323] Remove invite code from CD flow

### PR DESCRIPTION
**Pull request by ashley-weaver**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4323

## 🛠 Changes

Removes invite code from CD intake flow.

## ℹ️ Context

There are various, time-consuming hurdles CDs must jump through in order to accept an invitation for an organization (invite code, login.gov steps + identity proofing, PII checks). Given some CD users may be managing multiple organizations, we want to reduce friction in the accept invite workflow and remove the invite code step.

## 🧪 Validation
CD accept invite page no longer asks for verification code.
image

Invite flow goes to Login.gov step.
image

Tests updated.

### CODE REVIEW

File: dpc-portal/Gemfile.lock [hunk 1]
```diff
@@ -629,7 +629,6 @@ DEPENDENCIES
   view_component (~> 3.9)
   webdrivers
   webmock
-  webrick (~> 1.8.2)
   yard (>= 0.9.36)
```
Comment (jdettmannnava) [inline dpc-portal/Gemfile.lock:632]: How did this get removed without a change to the Gemfile?

File: dpc-portal/app/components/core/table/row_component.rb [hunk 2]
```diff
@@ -21,7 +21,6 @@ def initialize(obj:, obj_iteration:, keys:, delete_path: nil, obj_name: '')
       end
 
       def format_if_date(str)
-        # verification_code ABC123 parses
```
Comment (jdettmannnava) [inline dpc-portal/app/components/core/table/row_component.rb:24]: This comment explains why this check is being done

File: dpc-portal/app/components/page/invitations/start_component.html.erb [hunk 3]
```diff
@@ -53,12 +53,6 @@
             </li>
             <% end %>
           </ul>
-        </p>
```
Comment (jdettmannnava) [inline dpc-portal/app/components/page/invitations/start_component.html.erb:56]: I think this close tag is still necessary

File: dpc-portal/app/components/page/invitations/start_component.html.erb [hunk 4]
```diff
@@ -53,12 +53,6 @@
...
-        <p>
-          <% if @invitation.credential_delegate? %>
-            If you didn't receive this code contact <strong><%= @invitation.invited_by.given_name %> <%= @invitation.invited_by.family_name %></strong>. The code must not be sent via email.
-          <% end %>
-        </p>
```
Comment (jdettmannnava) [inline dpc-portal/app/components/page/invitations/start_component.html.erb:61]: The link below should no longer work. All the code-checking paths should also be removed from the routes.rb
This should require an update to the spec...

File: dpc-portal/spec/requests/invitations_spec.rb [hunk 5]
```diff
@@ -428,148 +428,6 @@
...
-        post "/organizations/#{org.id}/invitations/#{cd_invite.id}/verify_code", params: fail_params
-        expect(response).to be_forbidden
-        expect(response.body).to include(I18n.t('verification.max_tries_exceeded_text', ao_full_name: 'Gavin McCloud'))
-        expect(cd_invite.reload.failed_attempts).to eq 5
-        expect(cd_invite.unacceptable_reason).to eq 'max_tries_exceeded'
-      end
-    end
-  end
-
-  describe 'GET /confirm_cd' do
```
Comment (jdettmannnava) [inline dpc-portal/spec/requests/invitations_spec.rb:506]: Unsure why this test removed

**Review by jdettmannnava — COMMENTED** [review]
Looks good overall, though I think the start page goes to the code page now. I think you will need to run through it manually once just to verify.

File: dpc-portal/Gemfile.lock (hunk 1 above)
Comment (ashley-weaver) [reply dpc-portal/Gemfile.lock:632]: not sure, that's how it came out when I rebuilt

File: dpc-portal/app/components/page/invitations/start_component.html.erb (hunk 3 above)
Comment (ashley-weaver) [reply dpc-portal/app/components/page/invitations/start_component.html.erb:56]: my editor claimed it was connected to nothing, but I can see that's not true, oops.

File: dpc-portal/spec/components/page/invitations/start_component_spec.rb [hunk 6]
```diff
@@ -44,11 +44,5 @@
         expect(list_items[1]).to have_text('Enter your invite code')
       end
     end
-
-    it 'should go to code page' do
```
Comment (jdettmannnava) [inline dpc-portal/spec/components/page/invitations/start_component_spec.rb:48]: can we make sure it goes to confirm_cd?

File: dpc-portal/db/migrate/20241010163724_remove_verification_code_from_invitations.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class RemoveVerificationCodeFromInvitations < ActiveRecord::Migration[7.1]
+  def change
+    remove_column :invitations, :verification_code, :string
```
Comment (jdettmannnava) [inline dpc-portal/db/migrate/20241010163724_remove_verification_code_from_invitations.rb:3]: Also remove failed attempts?

**Review by jdettmannnava — COMMENTED** [review]
Looks good! Just some minor tweaks

**Review by jdettmannnava — APPROVED** [review]
LGTM

**Pull request merged**